### PR TITLE
Upgrade Crystal and Ameba

### PR DESCRIPTION
**Pull request by Sija**
- Updates Crystal to v1.8.2
- Updates Ameba to v1.4.3
- Removes `docs/description` directory on every (`src/generate.cr`) run, to ensure the obsolete files are gone
- Refactors the code in several places

### CODE REVIEW

**Comment by Sija** [comment]
ping @prcr @nicklem

File: docs/tests/Layout_TrailingBlankLines.cr [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # #Patterns: Layout_TrailingBlankLines
 
 puts "hello"
+
```
Comment (codacy-vrhpires) [inline docs/tests/Layout_TrailingBlankLines.cr:4]: Can you just remove this line and commit again please?

**Comment by Sija** [comment]
@codacy-vrhpires Is it GTG?

**Review by stefanvacareanu7 — APPROVED** [review]
LGTM!

**Comment by Sija** [comment]
I've fixed some things in the `README.md`, but now the markdown linter complains, and I have no idea how to please it properly, I'd appreciate an advice or a helpful hand :)

**Review by codacy-vrhpires — APPROVED** [review]
LGTM!

**Comment by Sija** [comment]
`publish_docker` job [failed](https://app.circleci.com/pipelines/github/codacy/codacy-ameba/118/workflows/e11f6a65-6c73-4f27-b272-e8ea09ba6cc1/jobs/424), although I don't think it's due to this PR.

**Comment by Sija** [comment]
btw, `version` in `shard.yml` should be probably bumped before release.

**Pull request merged**